lib/alertwildfire_API.py

```diff
@@ -65,22 +68,45 @@ def get_individual_camera_info(cameraID):
 def request_current_image(outputDir, cameraID):
     camera_info = get_individual_camera_info(cameraID)
     if camera_info["image"]["time"]:
-        timeStamp = camera["image"]["time"]###need to convert their format
+        timeStamp = camera_info["image"]["time"]###need to convert their format
+
+
+        logging.warning('not yet implemented camera_info/image/time')
+        timeStamp = time.mktime(datetime.datetime.now().timetuple())
+
+
     elif camera_info["position"]["time"]:
-        timeStamp = camera["position"]["time"]###need to convert their format
+        timeStamp = camera_info["position"]["time"]###need to convert their format
+
+
+        logging.warning('not yet implemented camera_info/image/time')
+        timeStamp = time.mktime(datetime.datetime.now().timetuple())
+
     else:
         timeStamp = time.mktime(datetime.datetime.now().timetuple())
     imgPath = img_archive.getImgPath(outputDir, cameraID, timeStamp)
     if os.path.isfile(imgPath):
         logging.warning('File %s already downloaded', imgPath)
         return imgPath
-    response = invokeApi("/currentimage", queryParams = "name="+cameraID, stream = True)
+    if camera_info["image"]["url"]:
+        url_override = camera_info["image"]["url"]
+    else:
+        url_override = False
+    response = invokeApi("/currentimage", queryParams = "name="+cameraID, stream = True, url_override = url_override)
     if response.status_code == 200:
         with open(imgPath, 'wb') as f:
             for chunk in response:
                 f.write(chunk)
             f.close()
         response.close()
+        f = open(imgPath, 'rb')
+        f_tags = exifread.process_file(f)
+        f.close()
+        if "Image DateTime" in f_tags.keys():
+            timeStamp = time.mktime(datetime.datetime.strptime(str(f_tags["Image DateTime"]), '%Y:%m:%d %H:%M:%S').timetuple())
+            newimgPath = img_archive.getImgPath(outputDir, cameraID, timeStamp)
+            os.rename(imgPath, newimgPath)
```

Comment:
I think rename fails if the two paths are the same, so please add a check to only call if they are different.

Reply:
the rename will not do anything if the paths are the same (tested ) so there is no need for a check